### PR TITLE
Clean up `pins` module

### PR DESCRIPTION
**Pull request by hannobraun**

### CODE REVIEW

File: src/pins/state.rs [hunk 1]
```diff
@@ -0,0 +1,67 @@
+//! Contains types that indicate pin states
+//!
+//! Please refer to [`Pin`] for documentation about how these types are used.
```
Comment (david-sawatzke) [inline src/pins/state.rs]: `Pin` isn't linked (compared to all the other links to `Pin` in this file)

File: src/pins/state.rs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+use core::marker::PhantomData;
+
+/// Implemented by types that indicate pin state
+///
+/// [`Pin`] uses this type as a trait bound for the type parameter that
+/// indicates the pin's state. This is done for the purpose of
+/// documentation, to show which states a pin can be in. Other than that,
+/// this trait should not be relevant to users of this crate.
+///
+/// [`Pin`]: ../struct.Pin.html
```
Comment (david-sawatzke) [inline src/pins/state.rs]: I think this link isn't valid anymore

Comment (david-sawatzke) [reply src/pins/state.rs]: (also applies to all the other links to `Pin`)

Comment (hannobraun) [reply src/pins/state.rs]: It is actually, due to the public re-export in `pins`. That's what I think anyway, but the link does work in the generated output.

**Comment by hannobraun** [comment]
I fixed all broken links in the `pins` documentation and updated some of it. Merging, since the previous commits were already approved, and the new commits only change documentation, which is largely outdated and in need of an update anyways.

**Pull request merged**